genpolicy: add --version flag

Redent0r opened the pull request:
Add a --version flag to the genpolicy tool that prints the current version

<!--
COMMENT BLOCKS WILL NOT BE INCLUDED IN THE PR.
Feel free to delete sections of the template which do not apply to your PR, or add additional details
-->

###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
<!-- **All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)* -->
- [x] Followed patch format from upstream recommendation: https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format
  - [x] Included a single commit in a given PR - at least unless there are related commits and each makes sense as a change on its own.
- [x] Aware about the PR to be merged using "create a merge commit" rather than "squash and merge" (or similar)
- [x] genPolicy only: Ensured the tool still builds on Windows
- [x] genPolicy only: Updated sample YAMLs' policy annotations, if applicable
- [x] The `upstream-missing` label (or `upstream-not-needed`) has been set on the PR. 

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of WHAT changed and WHY. -->
Add --version flag

`
$ target/debug/genpolicy -v

Kata Containers policy tool (Rust): id: genpolicy, version: 3.2.0-azl0.genpolicy1, commit: ffc72e513cee8b5686df20c04abf5cec77de8236
`

dev: instead of `cargo build`, we'd use `LIBC=gnu BUILD_TYPE= make` (`cargo build` still works, but you'll need to use `LIBC=gnu BUILD_TYPE= make` the first time at least to generate version.rs and not get a compile error)

release: instead of `cargo build --release`, we'd use `LIBC=gnu make` (Here we'll want to stick to `LIBC=gnu make` as this will embed the correct commit information to release binary)

Make build method is incompatible on Windows. For Windows releases:
- set `src/version.rs` with correct commit info
- `cargo build --release`

danmihai1 commented on src/tools/genpolicy/Cargo.toml:
```diff
@@ -5,7 +5,7 @@
 [package]
 name = "genpolicy"
-version = "0.1.0"
+version = "3.2.0-1"
```
Any chance we can do something closer to the way version gets handled upstream? E.g.,

/usr/local/bin/containerd-shim-kata-cc-v2 --version
Kata Containers containerd shim (Golang): id: "io.containerd.kata.v2", version: 3.3.0-alpha0, commit: 9b7dcc0d98cf138e519d65cac8ef52ae9f42a5fb

Redent0r replied:
Yea, I'll take a look

Redent0r replied:
Added

sprt commented on src/tools/genpolicy/Cargo.toml:
```diff
@@ -5,7 +5,7 @@
 [package]
 name = "genpolicy"
-version = "0.1.0"
+version = "3.2.0-1"
```
```suggestion
version = "3.2.0.azl0.genpolicy1"
```

Redent0r replied:
Version field will only accept semantic versioning. `3.2.0.azl0.genpolicy1` is not accepted by cargo
![image](https://github.com/microsoft/kata-containers/assets/30801614/d0fbed45-17f5-4101-8324-d75661ee17f2)

But funnily, it will take `3.2.0-azl0.genpolicy1` (with a dash instead of dot in the middle). Updating to that

sprt commented:
Also to update the release process in the wiki to bump the version number.

Redent0r commented:
Just force pushed following changes:
- Added commit info to version
- Separated `build` and `release` recipes. Then we can use `make`/`make build` for dev and `make release` when doing release, which seems more straightforward than using a unified recipe
-  Added license to Cargo.toml

Redent0r commented on src/tools/genpolicy/Makefile:
```diff
@@ -9,17 +9,37 @@ ifeq ($(ARCH), ppc64le)
 default: build
 
-build:
-	@RUSTFLAGS="$(EXTRA_RUSTFLAGS) --deny warnings" cargo build --target $(TRIPLE) --$(BUILD_TYPE)
```
This doesn't work in my machine. The error is coming from `--target $(TRIPLE)`. From https://doc.rust-lang.org/cargo/commands/cargo-build.html#compilation-options, target seems to be used to target different architectures. I don't think we need that for dev.

On the other hand, I do see how it's used in other makefiles like in https://github.com/microsoft/kata-containers/blob/msft-main/src/tools/kata-ctl/Makefile#L40, but I don't understand if it helps us. Maybe it's used in Kata CI? Found https://github.com/kata-containers/kata-containers/commit/201eec628a2dda2f92591eba1cb99c968ce819ea#diff-e157865d84d98087a8593904adb644a4d5348f59de4ca7111e9fedb9f406de70R97

Even if it is, we can leave this change in our fork and leave upstream makefile CI compatible

sprt replied:
I don't think we should change the behavior here as upstream builds using this target: https://github.com/kata-containers/kata-containers/blob/af3b19ed187e4412dee6ed96674714dd4d928422/tools/packaging/static-build/tools/build-static-tools.sh#L32

The problem is `$TRIPLE` gets evaluated to `x84_64-unknown-linux-musl` because `LIBC` defaults to musl in utils.mk. This works:

```
LIBC=gnu make
```

sprt commented on src/tools/genpolicy/src/version.rs.in:
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2020 Intel Corporation
```
Might want to add a line for Microsoft if we modified this file from Intel: https://docs.opensource.microsoft.com/contributing/

Redent0r replied:
Good catch. TIL about `Portions Copyright`

sprt reviewed (changes requested):
A couple comments. Also shouldn't we rather create this PR upstream?

Redent0r commented:
Addressed feedback and corrected commit message
